File: pinyougou-web/pinyougou-user-web/src/main/webapp/js/controller/userController.js
/** 定义控制器层 */
app.controller('userController', function($scope,$controller,$timeout,baseService){
    $controller('indexController',{$scope:$scope});
    $scope.user = {username:$scope.username};
    $scope.confirmPassword = "";
    $scope.code = "";
    $scope.checkTips = 1;
    $scope.tips = "";
    $scope.save = function () {
        if ($scope.confirmPassword && $scope.confirmPassword == $scope.user.password) {
                baseService.sendPost("/user/save?code=" + $scope.code,$scope.user).then(function (value) {
                    if (value.data) {
                        $scope.user = {};
                        $scope.confirmPassword = "";
                        $scope.code = "";
                        $scope.checkTips = 1;
                    } else {
                        alert("注册失败!");
                    }
                })
        } else {
            alert("两次输入密码不一致!");
        }
    };

    $scope.checkCode=function () {
        baseService.sendGet("/user/checkCode?code=" + $scope.code).then(function (value) {
            if (value.data) {
                $scope.isNext = value.data;
            } else {
                alert("验证码不正确!");
            }
        })
    }

    $scope.sendSms = function () {
        if ($scope.user.phone && /^1[3|4|5|7|8]\d{9}$/.test($scope.user.phone)) {
            baseService.sendGet("/user/sendSms?phone=" + $scope.user.phone);
            $scope.checkTips = 0;
            // 设置倒计时
            $scope.countDown(60);
        } else {
            alert("手机格式不正确!");
        }
    };

    $scope.countDown =function (count) {
        count --;
        $scope.tips = count +  "秒后重新获取!";
        if (count < 0) {
            $scope.checkTips = 1;
            return;
        }
        $timeout(function () {
            $scope.countDown(count);
        },1000)
    }

    $scope.changePw=function () {
        if($scope.user.newPassword == $scope.upsure && $scope.user.newPassword){
            if($scope.user.password){
                baseService.sendPost("/user/changePW",$scope.user).then(function (response) {
                    if (response.data) {
                        alert("修改密码成功！！！");
                        $scope.user=response.data;
                        $scope.user.password=null;
                        $scope.upsure="";
                    } else {
                        $("#originalPassword").html("原始密码错误");
                    }
                });
            }else{
                $("#originalPassword").html("原始密码不能为空");
            }
        }else{
            $("#newPassword").html("新的密码不能为空");
        }
    }
    
    $scope.showPhone=function () {
        baseService.sendGet("/user/showPhone").then(function (response) {
            $scope.user.username=null;
           $scope.user.phone=response.data;
        });
    }

    $scope.updatePhone=function () {
        if ($scope.user.phone && /^1[3|4|5|7|8]\d{9}$/.test($scope.user.phone)){
            $scope.user.username=null;
            baseService.sendPost("/user/updatePhone?picCode="+$scope.picCode+"&code="+$scope.code,
                $scope.user).then(function (response) {
                if(response.data){
                    location.href="/home-setting-address-complete.html";
                }else {
                    alert("验证码失败！！！");
                }
            })
        }
    }


});

File: pinyougou-web/pinyougou-user-web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.common.pojo.PageResult;
import com.pinyougou.pojo.*;
import com.pinyougou.service.*;
import com.pinyougou.pojo.Address;
import com.pinyougou.pojo.Cities;
import com.pinyougou.pojo.Provinces;
import com.pinyougou.pojo.User;
import com.pinyougou.service.*;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.util.DigestUtils;
import org.springframework.web.bind.annotation.*;

import javax.servlet.http.HttpServletRequest;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * UserController
 *
 * @version 1.0
 * @date 2019/4/15
 */
@RestController
@RequestMapping("/user")
public class UserController {

    @Reference(timeout = 10000)
    private UserService userService;
    @Reference(timeout = 10000)
    private ProvincesService provincesService;
    @Reference(timeout = 10000)
    private CitiesService citiesService;
    @Reference(timeout = 10000)
    private AreasService areasService;

    @Reference(timeout = 10000)
    private OrderService orderService;

    @Autowired
    private HttpServletRequest request;

    @Reference(timeout = 10000)
    private PayService payService;
    @Reference(timeout = 10000)
    private PayLogService payLogService;

    @Reference(timeout = 10000)
    private CartService cartService;

    @Reference(timeout = 10000)
    private AddressService addressService;


    @PostMapping("/save")
    public Boolean save(@RequestBody User user,String code) {
        try {
            if (userService.checkSmsCode(code)) {
                userService.save(user);
                return true;
            }
            return false;
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }

    @GetMapping("/sendSms")
    public Boolean sendSms(String phone) {
        try {
            return userService.sendSms(phone);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }

    @GetMapping("/showUser")
    public Map<String, Object> showUser() {
        Map<String, Object> map = new HashMap<>();
        String username = SecurityContextHolder.getContext().getAuthentication().getName();
        map.put("username", username);
        return map;
    }


    //获取用户订单信息
    @GetMapping("/getUserOrder")
    public PageResult getUserOrder(@RequestParam("page")int page,@RequestParam("rows") int rows ){

        try {
            //获取登录用户名
            String userId = request.getRemoteUser();
            return orderService.getUserOrder(page,rows,userId);
        }catch (Exception ex){
            ex.printStackTrace();
        }
        return null;
    }


    @GetMapping("/genPayCode")
    public Map<String, Object> genPayCode(String outTradeNo,String totalFee) {
        try {
            return payService.genPayCode(outTradeNo,totalFee);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }


    @GetMapping("/getStatus")
    public Map<String, Object> getStatus(String outTradeNo) {
        try {
            Map<String, Object> map = payService.getStatus(outTradeNo);
            return map;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }

    @GetMapping("/updatePayLog")
    public Boolean updatePayLog(String outTradeNo, String transactionId) {
        try {
            String userId = request.getRemoteUser();
            payLogService.updatePayLog(userId,outTradeNo,transactionId);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }

    @GetMapping("/findProvinces")
    public List<Provinces> findProvinces(){
        return provincesService.findAll();
    }

    @GetMapping("/findCitiesByParentId")
    public List<Cities> findCitiesByParentId(@RequestParam(value = "parentId") String parentId){
        return citiesService.findCitiesByParentId(parentId);
    }

    @GetMapping("/findAreasByParentId")
    public List<Areas> findAreasByParentId(@RequestParam(value = "parentId") String parentId){
        return areasService.findAreasByParentId(parentId);
    }

    @PostMapping("/saveOrUpdate")
    public Boolean saveOrUpdate(@RequestBody User user){
        try{
            String username = SecurityContextHolder.getContext().getAuthentication().getName();
            user.setUsername(username);
            userService.saveUserInfo(user);
            return true;
        }catch (Exception e){
            e.printStackTrace();
        }
        return false;
    }
    @GetMapping("/findUserInfo")
    public User findUserInfo(User user){
        String username = SecurityContextHolder.getContext().getAuthentication().getName();
        user.setUsername(username);
        User user1 = userService.findByUserName(user);
        return user1;
    }

    @GetMapping("/addPic")
    public Boolean addPic(String headPic){
        String username = SecurityContextHolder.getContext().getAuthentication().getName();
        return userService.addPicUrl(username,headPic);
    }


    //添加购物车
    @GetMapping("/addToCarts")
    public Boolean addToCarts(Long itemId, Integer num) {
        try {
            String userId = SecurityContextHolder.getContext().getAuthentication().getName();
            cartService.addItemToCartByRedis(userId, itemId, num);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }

    //获取地址
    @GetMapping("/getAddress")
    public List<Map<String,Object>> getAddress() {
        try {
            String userId = request.getRemoteUser();
            return addressService.findByUserName(userId);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return null;
    }

    //更新地址
    @PostMapping("/updateAddress")
    public Boolean update(@RequestBody Address address){
        try {
            String userId = request.getRemoteUser();
            address.setUserId(userId);
            addressService.update(address);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }

    //保存地址
    @PostMapping("/saveAddress")
    public Boolean save(@RequestBody Address address){
        try {
            String userId = request.getRemoteUser();
            address.setUserId(userId);
            addressService.save(address);
            return true;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }


    @GetMapping("/deleteAddress")
    public boolean deleteAddress(Long id){
        try {
            addressService.delete(id);
            return true;
        }catch (Exception ex){
            ex.printStackTrace();
        }
        return false;
    }


    @GetMapping("/setDefaultAddress")
    public boolean setDefaultAddress(Long id,HttpServletRequest request){
        try {
            String userId = request.getRemoteUser();
            addressService.setDefaultAddress(id,userId);
            return true;
        }catch (Exception ex){
            ex.printStackTrace();
        }
        return false;
    }



    //修改密码
    @PostMapping("/changePW")
    public User changePW(@RequestBody Map<String, Object> userMap, HttpServletRequest request){
        String password = (String) userMap.get("password");
        String pass1 = DigestUtils.md5DigestAsHex(password.getBytes());
        String username = request.getRemoteUser();
        User user1=userService.selectUser(username);

        String oldPassword = user1.getPassword();
        if(oldPassword.equals(pass1)){
            userService.updatePassWord((String) userMap.get("newPassword"),username);
            user1.setPassword(null);
            return user1;
        }
        return null;
    }

    //根据用户名或取手机号码
    @GetMapping("showPhone")
    public String showPhone(HttpServletRequest request){
        String username = request.getRemoteUser();
        User user1 =userService.selectUser(username);
        String phone = user1.getPhone();
        return phone;
    }

    @GetMapping("/checkCode")
    public Boolean checkCode(String code){
       return userService.checkSmsCode(code);
    }
    @PostMapping("/updatePhone")
    public Boolean updatePhone(String picCode,String code,@RequestBody User user,HttpServletRequest request){
        try {
            String username = request.getRemoteUser();
            String verify_code = (String) request.getSession().getAttribute("verify_code");
            if(verify_code.equals(picCode)){
                if (userService.checkSmsCode(code)){
                    user.setUsername(username);
                    userService.update(user);
                    return true;
                }else{
                    return false;
                }
            }else{
                return false;
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return false;
    }

    @GetMapping("/showPic")
    public Map<String, String>  showPic(HttpServletRequest request){
        String username = request.getRemoteUser();
        User user = userService.selectUser(username);
        Map<String,String> map=new HashMap<>();
        map.put("headPic",user.getHeadPic());
        return map;
    }
}
